From 09cf89600e3904c32d7f70ce10e753b6794ed5a2 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 13 Sep 2021 10:15:36 +0300
Subject: [PATCH] [SW-0] greco: pci: disable mem access during iatu init

Experience with gaudi shows that completely blocking pci access
during iATU init can cause issues with the host, and it's against
the pci spec.
So to be on the safe side, only the memory access is disabled
during the iATU initialization.

Change-Id: Ifcf610e49c583f12f9b649805656a8caa3ae44c3
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/greco/common/asic.c b/arch/greco/common/asic.c
index 95fc5a3..9bb0e4a 100644
--- a/arch/greco/common/asic.c
+++ b/arch/greco/common/asic.c
@@ -57,7 +57,7 @@
 	struct hl_pci_inbound_region inbound_region;
 	struct hl_pci_outbound_region outbound_region;
 
-	hl_pcie_disable_access();
+	hl_pcie_disable_mem_access();
 
 	/* Inbound Region 0 - Bar - - Point to SRAM and CFG */
 	inbound_region.mode = PCI_BAR_MATCH_MODE;
@@ -76,7 +76,7 @@
 	outbound_region.size = HOST_PHYS_SIZE;
 	hl_pci_set_outbound_region(&outbound_region);
 
-	hl_pcie_enable_access();
+	hl_pcie_enable_mem_access();
 }
 
 void lkd_dev_spec_regs(struct cpu_dyn_regs *regs)
